lexoid/core/parse_type/static_parser.py

```diff
@@ -233,46 +286,42 @@ def detect_heading_level(font_size):
                 # If we were collecting a heading
                 if current_heading:
                     level = detect_heading_level(current_heading[0]["size"])
-                    heading_text = " ".join(word["text"] for word in current_heading)
-                    markdown_content.append(f"{'#' * level} {heading_text}\n\n")
+                    heading_text = format_paragraph(current_heading)
```

Comment:
Before after for `test_explicit_hyperlink_n_img.pdf`

Before:
![Screenshot 2024-12-27 at 11 05 36 AM](https://github.com/user-attachments/assets/96085ecf-94d8-4206-87cd-ec89edf9b3bd)

After:
![Screenshot 2024-12-27 at 10 49 44 AM](https://github.com/user-attachments/assets/84e511f4-ee13-43d5-bdf5-25364a05f16b)